### PR TITLE
EOS-21195: Feature audit logs

### PR DESCRIPTION
**Pull request by soniyamoholkar**
# Problem Statement
- Problem statement 1. Sorting is not working with text fields in ES. 2. s3-rsys-index should not be generated through generic DB framework.

# Design
-  For Bug describe the fix here. 
-  For Feature, Post the link to the solution page on the confluence CORTX Foundation Library 
- https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/494600216/Rsyslog+and+Elasticsearch
- https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/486114207/Elasticsearch-+Sorting+on+text+fields

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: 
- [ ] Confirm Testing was performed with installed RPM [Y/N]:  

# Review Checklist 
  Before posting the PR please ensure
- [ ] PR is self reviewed
- [ ] Is there a change in filename/package/module or signature [Y/N]: 
- [ ] If yes for above point, Is a notification sent to all other cortx components [Y/N]
- [ ] Jira is updated
- [ ] Check if the description is clear and explained. 
- [ ] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/utils/data/db/consul_db/storage.py [hunk 1]
```diff
@@ -229,18 +229,19 @@ def __init__(self, consul_client: Consul, model: Type[BaseModel],
         self._model_scheme = dict()
 
     @classmethod
-    async def create_database(cls, config, collection: str,
+    async def create_database(cls, config, model_settings,
```
Comment (ujjwalpl) [inline py-utils/src/utils/data/db/consul_db/storage.py:232]: - Why is there no type associated with model_settings. 
- Adding variable in the middle without default value will break the backward compatibility. Need to find all the uses and ensure call is changed to include new interface.

File: py-utils/src/utils/data/db/elasticsearch_db/storage.py [hunk 2]
```diff
@@ -331,17 +337,18 @@ def __init__(self, es_client: Elasticsearch, model: Type[BaseModel], collection:
...
 
     @classmethod
-    async def create_database(cls, config, collection, model: Type[BaseModel]) -> IDataBase:
+    async def create_database(cls, config, model_settings, model: Type[BaseModel]) -> IDataBase:
         """
         Creates new instance of ElasticSearch DB and performs necessary initializations
 
         :param DBSettings config: configuration for elasticsearch server
-        :param str collection: collection for storing model onto db
+        :param :param model_settings: Model settings
```
Comment (ujjwalpl) [inline py-utils/src/utils/data/db/elasticsearch_db/storage.py:345]: Description is not clear enough to know what is model settings.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/utils/data/db/consul_db/storage.py (hunk 1 above)
Comment (soniyamoholkar) [reply py-utils/src/utils/data/db/consul_db/storage.py]: Addressed the comments, added new optional parameter create_schema with default value as True. Please check

File: py-utils/src/utils/data/db/elasticsearch_db/storage.py (hunk 2 above)
Comment (soniyamoholkar) [reply py-utils/src/utils/data/db/elasticsearch_db/storage.py]: Updated the comments. Please check

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/utils/data/db/db_provider.py [hunk 3]
```diff
@@ -158,7 +159,8 @@ async def create_database(self) -> None:
         try:
             self._database = await self._database_module.create_database(self._db_config.config,
                                                                          self._model_settings.collection,
-                                                                         self._model)
+                                                                         self._model,
```
Comment (sachinpunadikar) [inline py-utils/src/utils/data/db/db_provider.py:162]: indentation do not look good in browser, please cross check

Comment (soniyamoholkar) [reply py-utils/src/utils/data/db/db_provider.py]: Addressed the review comment. Please check once.

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**